src/main/java/lotto/step2/domain/Lotto.java

```diff
@@ -0,0 +1,39 @@
+package lotto.step2.domain;
+
+import java.util.*;
+
+public class Lotto {
+    private final static int MATCHED = 1;
+    private final static int MISMATCHED = 0;
+    private final static String ILLEGAL_LOTTO = "로또는 6개의 서로 다른 숫자로 만들어져야 합니다";
+    private Set<LottoNumber> lottoNumbers = new TreeSet<>();
+
+    public Lotto(List<LottoNumber> lottoNumbers) {
+        this.lottoNumbers = new TreeSet<>(lottoNumbers);
+        validateLottoSize();
+    }
+
+    public Lotto(String[] numbers) {
+        for(String number : numbers) {
```

Comment:
```suggestion
    public Lotto(List<String> numbers) {
        for(String number : numbers) {
```

다음 미션의 사다리타기를 위해서라도, 배열을 쓰지 않고 전부 List 인터페이스로 통일해보는 연습은 어떠신가요?
참고로 이건 다음 미션을 대비한 추천일뿐, 꼭 변경하지 않으셔도 괜찮습니다.
```suggestion
    public Lotto(String[] numbers) {
        for(String number : numbers) {
```

Reply:
배열에 비해 리스트의 장점이 삽입, 삭제가 용이하다는 점 같은데 사다리타기 미션에서 삽입 삭제가 많이 일어나서 그런가요?

Reply:
아뇨 Collections 관련 기능들을 더 적극적으로 활용해보기를 바라는 마음에서 그렇습니다.
for loop 와 Collections 기능을 잘 못쓰면서 Stream 에만 집착하시는 분들을 너무 많이 목격해서요.

Reply:
보면 제가 이미 public Lotto(List<LottoNumber> lottoNumbers) 생성자를 갖고 있어서  public Lotto(List<String> numbers 
생성할 수가 없네요 ㅠㅠ
그래서 일단 String배열로 해놓았습니다
public Lotto(List<LottoNumber> lottoNumbers) 생성자를 버리는게 나을까요?
객체를 주고 받는게 좋다고 생각해서 제 눈에는 예뻐(?) 보이는 생성자였는데 고민되네요...

Reply:
same erasure 에러가 뜨는 군요.
이런 경우에는 저는 static 생성자 메서드를 활용하면 어떨까 제안을 해봅니다.
다만, 이건 이펙티브 자바에서 권고하는 방법이고,
"엘레강트 오브젝트" 관점에서은 static 메서드를 극혐하므로, 꼭 따를 필요는 없습니다.